components/operator/internal/state/served_filter.go

```diff
@@ -32,6 +32,7 @@ func setInitialServed(ctx context.Context, r *reconciler, s *systemState) error
 
 func setServed(servedServerless *v1alpha1.Serverless, s *systemState) error {
 	if servedServerless == nil {
+		s.setState(v1alpha1.StateProcessing)
```

Comment:
Shouldn't we check `State` before updating?
IMO. We should update only empty `State`.

Reply:
I've removed the `required` status from the `State` field to restore old behavior; this change is no longer needed